kolibri/core/tasks/api.py

```diff
@@ -207,8 +244,50 @@ def list(self, request):
         return Response(jobs_response)
 
     def create(self, request):
-        # unimplemented. Call out to the task-specific APIs for now.
-        pass
+        """
+        Enqueue a task for async processing.
+
+        API endpoint:
+            POST /api/tasks/
+
+        Request payload parameters:
+            - `task` (required): a string representing the dotted path to task function.
+            - all other key value pairs are passed to the validator if the
+              task function has one otherwise they are passed to the task function itself
+              as keyword args.
+
```

Comment:
We don't need to address this in this PR, but one thought that occurs is that it would be useful to be able to schedule tasks through this endpoint as well.

Any thoughts on how we could extent the API to allow for that? Possibly optional `enqueue_at` and `enqueue_in` arguments?

Reply:
Yes, we can extent our API to make it do that. 

Not only `enqueue_at` and `enqueue_in` but also `repeat` & `interval`. 

So, here we have two choices of request payloads:

**1.** 
```
{
  "task": "....",
  "enqueue_at": "July 27 2021  2:00PM",
  "repeat": 5,
  "interval": 86400, (in seconds)
  ... (args that get passed to the task function)
}
```  
**2.**
```
{
  "task": "....",
  "enqueue_at": {
        "time": "July 27 2021  2:00PM",
        "repeat": 5,
        "interval": 86400, (in seconds)
       }
  ... (args that get passed to the task function)
}
```  

My choice is the second one as it offers zero chance of conflict with names of task function arguments. What's your say?

Reply:
I do prefer the second option, yes - there are some other options, but this seems the most sensible.